flecsi/utils/mpi_type_traits.h

```diff
@@ -121,7 +121,7 @@ struct mpi_typetraits_u {
       return MPI_CXX_DOUBLE_COMPLEX;
     else if constexpr(is_same_v<TYPE, complex<long double>>)
       return MPI_CXX_LONG_DOUBLE_COMPLEX;
-    else if constexpr(is_same_v<TYPE, byte>)
+    else if constexpr(is_same_v<TYPE, unsigned char>)
```

Comment:
`unsigned char` is [already handled above](laristra/flecsi/blob/0d0f74fc2542320761b33f563193e28747031ae1/flecsi/utils/mpi_type_traits.h#L65); why would we want it to be `MPI_BYTE` instead?

Reply:
Fair point. I'll go back and check why I thought that this was needed.

Reply:
This has been changed to refer to `std::byte` instead in #645.